02_configure_host.sh

```diff
@@ -198,6 +200,15 @@ elif [[ "$reg_state" != "running" ]]; then
 fi
 sleep 5
 
+
+# Clone all needed repositories (CAPI, CAPM3, BMO, IPAM)
+mkdir -p "${M3PATH}"
+clone_repo "${BMOREPO}" "${BMOBRANCH}" "${BMOPATH}" "${BMOCOMMIT}"
+clone_repo "${CAPM3REPO}" "${CAPM3BRANCH}" "${CAPM3PATH}" "${CAPM3COMMIT}"
+clone_repo "${IPAMREPO}" "${IPAMBRANCH}" "${IPAMPATH}" "${IPAMCOMMIT}"
+clone_repo "${CAPIREPO}" "${CAPIBRANCH}" "${CAPIPATH}" "${CAPICOMMIT}"
+
```

Comment:
Keep in mind that 02 script was meant for configuring host and 03 for mgmt cluster. With this change , you would be taking a part of the  management cluster work here in 02. So the naming of the scripts would need to change perhaps? Also, how would it affect [make  delete_mgmt_cluster](https://github.com/metal3-io/metal3-dev-env/blob/main/Makefile#L17) and [make launch_mgmt_cluster](https://github.com/metal3-io/metal3-dev-env/blob/main/Makefile#L9) commands, cause currently it is only related to 03 script.

Reply:
It simply clones repos to the host, the mgmt cluster customization still happens later, the repos are only used in potential local buildings in 02 nothing more. IMO this action should have been in  02 not in 03. 

No effect on cleanup this cloning is self contained as the clone_repo function handles the cleanup based on the FORCE_REPO_UPDATE var and this process is independent from the cluster cleanup.

Reply:
To rephrase myself, repo cloning  IMO is not part of mgmt kustomization and lunching just as much as libvirt vm creation is not part of it. These actions should be completed before  we launch mgmt and not during.

Reply:
ok so I would be able to delete and launch mgmt cluster from the same state/settings irrespective of this as it is now. I agree with cloning should be a part of 02 since it is where image is being built.